### ISSUE TITLE
Facets are been deprecated in Elasticsearch 2.0

### ISSUE THREAD

**Issue by SalahAdDin** [django-oscar/django-oscar#2089]
Hi guys, i'm working with `haystack 2.5` and `elasticsearch 2.5.3` and i have this [bug](https://github.com/django-haystack/django-haystack/issues/1247#issuecomment-238462589).

This happen because ElasticSearch deprecated `facets` and doesn't matters if `haystack` branches support now the lastes ElasticSearch version, Oscar use faces for filters.

So, what solution we have here?

Is better use the latest Sorl version?

**Comment by Alkalit** [comment]
In my current project im using haystack and elasticsearch 2.5. I used this branch https://github.com/joaojunior/django-haystack/tree/es2_refactoring
for handle with facets/aggs and it work well.

**Comment by SalahAdDin** [comment]
Have you a few steps for working well?
Have i edit something in `django-oscar` or `django-haystack` code?

**Comment by NDevox** [comment]
For those interested I ported the current elasticsearch engine for haystack over to 2.X. I haven't managed to get it working with 5.x yet.

You can find it here: https://pypi.python.org/pypi/elasticsearch2-haystack

It should work for all versions up to 2.4 but it isn't perfect as it essentially patches 2.x to look like 1.x.

From what I've heard there is someone working on a newer backend which is hopefully optimised for the latest releases of ES which this isn't as far as I'm concerned. So I'm not going to make a PR for haystack - I'm just making it available until there is a better replacement.

We've tested it at our work on Django-Oscar and it works just fine right now.

**Comment by SalahAdDin** [comment]
Which steps for install it?

**Comment by NDevox** [comment]
@SalahAdDin

There are some details on the pypi/github page (https://github.com/ndevox/elasticsearch2-haystack)

Essentially:

- Pip install the package (`pip install elasticsearch2-haystack`).
- Check which elasticsearch version you are running.
- Make sure you install the version of elasticsearch-py which matches the version of elasticsearch you are using.
- Add the engine to your `HAYSTACK_CONNECTION` setting like so:
```
HAYSTACK_CONNECTIONS = {
    'default': {
        'ENGINE': 'elasticsearch2_backend.ElasticsearchSearchEngine',
        'URL': 'http://127.0.0.1:9200/',
        'INDEX_NAME': 'haystack',
    },
}
```

After that it should be ready.

**Comment by SalahAdDin** [comment]
@NDevox, what's about ElasticSearch 5.X.X?

**Comment by NDevox** [comment]
@SalahAdDin I haven't had time to add 5.x support as we use AWS at work which only supports 2.3 right now. As a result 2.x had to be my focus.

**Comment by SalahAdDin** [comment]
Ah, if is the case, ok, but i use docker for use elastic search.

**Comment by NDevox** [comment]
You can still use 2.x with docker. Just use elasticsearch:2 as the image.

**Comment by SalahAdDin** [comment]
Thanks! Maybe after can work in add 5.x support, but, i think, first is needed update this package.

**Comment by solarissmoke** [comment]
This will be addressed in #2539
